src/validation.cpp

```diff
@@ -211,14 +211,75 @@ bool CheckFinalTx(const CBlockIndex* active_chain_tip, const CTransaction &tx, i
     return IsFinalTx(tx, nBlockHeight, nBlockTime);
 }
 
+bool CalculateLockPoints(
+    CBlockIndex* tip,
+    const CCoinsView& coins_view,
+    const CTransaction& tx,
+    int flags,
+    LockPoints* lp)
```

Comment:
It would be nice to remove the raw pointers while we're changing the interface anyway. I think a better interface is to have `CalculateLockPoints()` return a `std::optional<LockPoints>` which encodes the same information as the bool.

Reply:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/23897#issuecomment-1102831191).